cluster/repository/identity_cluster_blackbox_test.go

```diff
@@ -0,0 +1,129 @@
+package repository_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-cluster/cluster/repository"
+	"github.com/fabric8-services/fabric8-cluster/gormtestsupport"
+	"github.com/fabric8-services/fabric8-cluster/test"
+	"github.com/fabric8-services/fabric8-common/errors"
+
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type identityClusterTestSuite struct {
+	gormtestsupport.DBTestSuite
+	repo repository.IdentityClusterRepository
+}
+
+func TestIdentityCluster(t *testing.T) {
+	suite.Run(t, &identityClusterTestSuite{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *identityClusterTestSuite) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.repo = repository.NewIdentityClusterRepository(s.DB)
+}
+
+func (s *identityClusterTestSuite) TestCreateAndListIdentityClusterOK() {
+	// Create two identities for the same cluster
+	idCluster1 := test.CreateIdentityCluster(s.T(), s.DB, nil, nil)
+	idCluster2 := test.CreateIdentityCluster(s.T(), s.DB, nil, &idCluster1.IdentityID)
+
+	// Noise
+	test.CreateIdentityCluster(s.T(), s.DB, &idCluster1.Cluster, nil)
+	test.CreateIdentityCluster(s.T(), s.DB, nil, nil)
+
+	clusters, err := s.repo.ListClustersForIdentity(context.Background(), idCluster1.IdentityID)
+	require.NoError(s.T(), err)
+	assert.Len(s.T(), clusters, 2)
+	assertContainsCluster(s.T(), clusters, &idCluster1.Cluster)
+	assertContainsCluster(s.T(), clusters, &idCluster2.Cluster)
+}
+
+func (s *identityClusterTestSuite) TestListClustersForUnknownIdentityOK() {
+	clusters, err := s.repo.ListClustersForIdentity(context.Background(), uuid.NewV4())
+	require.NoError(s.T(), err)
+	assert.Len(s.T(), clusters, 0)
+}
+
+func assertContainsCluster(t *testing.T, clusters []repository.Cluster, cluster *repository.Cluster) {
+	require.NotEqual(t, uuid.UUID{}, cluster.ClusterID)
+	for _, cls := range clusters {
+		if cls.ClusterID == cluster.ClusterID {
+			return
+		}
+	}
+	assert.Fail(t, "didn't find cluster")
```

Comment:
How about giving some hint about `clusterID` in failure message?

Reply:
Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/e81043f2775af1cd77a47a8671ff304baf49e83c